src/components/filter/filter-amount/FilterAmount.tsx

```diff
@@ -0,0 +1,156 @@
+import { type Dispatch, useEffect, useState } from 'react';
+
+import styled from 'styled-components';
+
+import {
+  type FilterAmountAction,
+  type FilterAmountState,
+  filterAmonutOptions,
+} from 'src/components/filter/filter-amount/filterAmountReducer';
+import {
+  type BaseFilterProps,
+  useFilterWrapper,
+} from 'src/components/filter/useFilterWrapper';
+import { Input } from 'src/components/input/Input';
+import { Select } from 'src/components/select/Select';
+import { ArrowRightIcon } from 'src/icons/ArrowRightIcon';
+import { theme } from 'src/styles/constants/theme';
+
+const InputWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  gap: ${theme.space8};
+
+  svg {
+    min-width: 24px;
+  }
+`;
+
+export type FilterAmountType = 'equal' | 'between' | 'greater' | 'less';
+
+type FilterAmountProps = BaseFilterProps & {
+  dispatch: Dispatch<FilterAmountAction>;
+  filter: FilterAmountState;
+};
+
+export const FilterAmount = ({
+  dispatch,
+  dropdownTitle,
+  filter,
+  label,
+}: FilterAmountProps) => {
+  const [editFilterBy, setEditFilterBy] = useState<FilterAmountType | null>(
+    filter.amountFilterType,
+  );
+  const [editingAmount1, setEditingAmount1] = useState<number | null>(null);
+  const [editingAmount2, setEditingAmount2] = useState<number | null>(null);
+  const [newFilterText, setNewFilterText] = useState('');
+
+  const dispatchAmountFilterType = (value: FilterAmountType | null) => {
+    dispatch({
+      name: 'amountFilterType',
+      type: 'change',
+      value,
+    });
+  };
+
+  const dispatchAmounts = (values: {
+    amount1: number | null;
+    amount2: number | null;
+  }) => {
+    dispatch({
+      name: 'amountTotalMin',
+      type: 'change',
+      value: values.amount1,
+    });
+    dispatch({
+      name: 'amountTotalMax',
+      type: 'change',
+      value: values.amount2,
+    });
+  };
+
+  const handleApply = () => {
+    dispatchAmountFilterType(editFilterBy);
+    const amount1String = `${editingAmount1?.toFixed(2) || 0}` || '';
+    const amount2String = `${editingAmount2?.toFixed(2) || 0}` || '';
+
+    switch (editFilterBy) {
+      case 'between':
+        dispatchAmounts({ amount1: editingAmount1, amount2: editingAmount2 });
+        setNewFilterText(`Between ${amount1String} and ${amount2String}`);
+        break;
+      case 'equal':
+        dispatchAmounts({ amount1: editingAmount1, amount2: editingAmount1 });
+        setNewFilterText(`Equal to ${amount1String}`);
+        break;
+      case 'greater':
+        dispatchAmounts({ amount1: editingAmount1, amount2: null });
+        setNewFilterText(`Greater than ${amount1String}`);
+        break;
+      case 'less':
+        dispatchAmounts({ amount1: null, amount2: editingAmount1 });
+        setNewFilterText(`Less than ${amount1String}`);
+        break;
+      default:
+        // Handle default case if necessary
```

Comment:
```suggestion
```
I would think this is self explanatory 🤷🏼‍♂️ Sorry if you copied this from me lol